x/authz/keeper/keeper.go

```diff
@@ -84,10 +84,20 @@ func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []
 
 		// if granter != grantee then check authorization.Accept, otherwise we implicitly accept.
 		if !granter.Equals(grantee) {
-			authorization, _ := k.GetCleanAuthorization(ctx, grantee, granter, sdk.MsgTypeURL(msg))
-			if authorization == nil {
+			grant, found := k.getGrant(ctx, grantStoreKey(grantee, granter, sdk.MsgTypeURL(msg)))
+			if !found {
 				return nil, sdkerrors.ErrUnauthorized.Wrap("authorization not found")
 			}
+
+			if grant.Expiration.Before(ctx.BlockHeader().Time) {
+				return nil, sdkerrors.ErrUnauthorized.Wrap("authorization expired")
+			}
+
+			authorization := grant.GetAuthorization()
```

Comment:
how possible this can be `nil`? Let 's update the message below to be more clear that there is a logic error if this happens.